src/test/scala/com/fulcrumgenomics/fastq/DemuxFastqsTest.scala

```diff
@@ -0,0 +1,336 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.fastq
+
+import java.nio.file.Files
+
+import com.fulcrumgenomics.testing.{ErrorLogLevel, SamRecordSetBuilder, UnitSpec}
+import com.fulcrumgenomics.util.{Io, Metric, ReadStructure, SampleBarcodeMetric}
+import com.fulcrumgenomics.FgBioDef.FilePath
+import com.fulcrumgenomics.fastq.DemuxFastqs.UnmatchedSampleId
+import com.fulcrumgenomics.fastq.FastqDemultiplexer.DemuxRecord
+import org.scalatest.OptionValues
+import com.fulcrumgenomics.util.miseq.{Sample, SampleBarcode, SampleSheet}
+import dagr.commons.io.PathUtil
+import htsjdk.samtools.SAMUtils
+
+import scala.collection.mutable.ListBuffer
+
+class DemuxFastqsTest extends UnitSpec with OptionValues with ErrorLogLevel {
```

Comment:
@tfenne I really like both the `OptionValues` and `ErrorLogLevel` traits.  The former allows us to replace the `get` method with the `value` method for `Option`s and scalatest report a meaningful error when the `Option` is `None`.  The latter allows us to turn off logging to the console when running end-to-end `tool.execute()` tests.